python/lance/types/image.py

```diff
@@ -218,3 +263,70 @@ class ImageUriScalar(pa.ExtensionScalar):
 
     def as_py(self) -> Image:
         return ImageUri(self.value.as_py())
+
+
+class ImageArray(pa.ExtensionArray):
+    @staticmethod
+    def from_pandas(obj, mask=None, type=None, safe=True, memory_pool=None):
+        if isinstance(obj, pa.ChunkedArray):
+            chunks = [ImageArray.from_pandas(c) for c in obj.chunks]
+            return pa.chunked_array(chunks, chunks[0].type)
+
+        if isinstance(obj, pa.Array):
+            if pa.types.is_binary(obj.type):
+                return ImageArray.from_storage(ImageBinary.DTYPE, obj)
+            elif pa.types.is_string(obj.type):
+                return ImageArray.from_storage(ImageUri.DTYPE, obj)
+
+        if isinstance(obj, (list, tuple, np.ndarray)) and len(obj) > 0:
+            first = obj[0]
+            if isinstance(first, Image):
+                return ImageArray.from_images(obj, mask, type, safe, memory_pool)
+            elif isinstance(first, bytes):
+                storage = pa.array(obj, mask=mask, type=pa.binary(),
+                                   safe=safe, memory_pool=memory_pool)
+                return ImageArray.from_pandas(storage)
+            elif isinstance(first, str):
+                storage = pa.array(obj, mask=mask, type=pa.string(),
+                                   safe=safe, memory_pool=memory_pool)
+                return ImageArray.from_pandas(storage)
+
+        return pa.ExtensionArray.from_pandas(
+            obj, mask=mask, type=type, safe=safe, memory_pool=memory_pool
+        )
+
+    @staticmethod
+    def from_images(images, type=None, mask=None, safe=True, memory_pool=None):
+        """
+        Create an ImageUriArray from Image instances
+        Parameters
+        ----------
+        images : sequence of Image
```

Comment:
does it expect list / set / tuple or pd.Series / np.array?

Reply:
anything that's list-like. I can change this to "sequence / iterable / ndarray / pandas.Series of Image" to be more explicit (matches pa.array docstring)